src/ILCompiler.WebAssembly/src/CodeGen/WebAssemblyObjectWriter.cs

```diff
@@ -217,6 +213,10 @@ public static LLVMValueRef EmitGlobal(LLVMModuleRef module, FieldDesc field, Nam
                     var llvmValue = LLVM.AddGlobal(module, valueType, nameMangler.GetMangledFieldName(field).ToString());
                     LLVM.SetLinkage(llvmValue, LLVMLinkage.LLVMInternalLinkage);
                     LLVM.SetInitializer(llvmValue, GetConstZeroArray(field.FieldType.GetElementSize().AsInt));
+                    if (field.IsThreadStatic)
+                    {
+                        LLVM.SetThreadLocal(llvmValue, LLVMMisc.True);
```

Comment:
It's cool that LLVM can do this for us 😄. What happens with initialization for this? Can you please also add a test using a thread local? I know we can't test it multithreaded, but it would be good to just ensure reading/writing works right.

Reply:
@hippiehunter what I meant about initialization is that .NET thread static fields are supposed to be zero/null initialized. Does LLVM do that for us or do you also need to do some sort of initialization?

Reply:
I've added a test to validate the initial value of zero and that works. My understanding is that this should all be managed by the call to SetInitializer directly above.